runners/samza/src/main/java/org/apache/beam/runners/samza/translation/SamzaTestStreamTranslator.java

```diff
@@ -90,11 +95,53 @@ public void translate(
     ctx.registerInputMessageStream(output, inputDescriptor);
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public void translatePortable(
       PipelineNode.PTransformNode transform,
       QueryablePipeline pipeline,
       PortableTranslationContext ctx) {
-    throw new SamzaException("TestStream is not supported in portable by Samza runner");
+    final String outputId = ctx.getOutputId(transform);
+    final String escapedOutputId = SamzaPipelineTranslatorUtils.escape(outputId);
+    final GenericSystemDescriptor systemDescriptor =
+        new GenericSystemDescriptor(escapedOutputId, SamzaTestStreamSystemFactory.class.getName());
+    final ByteString bytes = transform.getTransform().getSpec().getPayload();
+    final RunnerApi.TestStreamPayload payload;
+    final Coder<T> coder;
+
+    try {
+      payload = RunnerApi.TestStreamPayload.parseFrom(bytes);
+      coder =
+          (Coder<T>)
+              RehydratedComponents.forComponents(pipeline.getComponents())
+                  .getCoder(payload.getCoderId());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    // the decoder for encodedTestStream
+    SerializableFunction<String, TestStream<?>> testStreamDecoder =
+        string -> {
+          try {
+            return TestStreamTranslation.testStreamFromProtoPayload(payload, coder);
+          } catch (IOException e) {
+            throw new SamzaException("Could not decode TestStream.", e);
+          }
+        };
+
+    final Map<String, String> systemConfig =
+        ImmutableMap.of(
+            "encodedTestStream",
+            Base64Serializer.serializeUnchecked(bytes),
```

Comment:
Is this being used? Seems the decoder function is already serialized with payload and actual coder, so not sure whether we need to pass in it here.

Reply:
Updated the decoder function not to encode the input anymore.